### PR TITLE
Migrate to uv for package management

### PR DESCRIPTION
**Pull request by munkyshi**
### Linked issue(s)

KOL-7307

### What change does this PR introduce and why?

Replace `poetry` with `uv` for dependency management and tooling.

Installing `kolena` is potentially a long process, and the dependency resolution time with `poetry` has crept up to the point of causing our workflow tests to flake. `uv` seems to offer a much faster interface for this, while preserving simplicity of development.

Also removes codecov for now, as it's something that hasn't been actively used / monitored.

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

**Comment by grant-Kolena** [comment]
Nice! Glad to see the build is working again

**Review by grant-Kolena — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -23,61 +25,37 @@ classifiers = [# classifiers for license, versions set automatically during Poet
...
-pytest-depends = "^1.0.1"
-mkdocs = "^1.4.3"
-cairosvg = "^2.7.0"
-mkdocs-material = ">=9.2,<10"  # insiders fork installed out-of-band in docs/setup_insiders.sh
-mkdocstrings = { version = ">0.20,<1", extras = ["python"] }
-mkdocs-git-committers-plugin-2 = "^1"
-mkdocs-git-revision-date-localized-plugin = "^1"
-
-[tool.poetry.scripts]
-kolena = 'kolena._utils.cli:run'
```
Comment (yifanwu-kolena) [inline pyproject.toml]: i think we need to find the counterpart of uv, b/c it provides a kolena cli in the python venv after installing kolena package

Comment (munkyshi) [reply pyproject.toml]: @yifanwu-kolena what is this used for / who uses it (is it something publicly shared / documented)? I couldn't find a great counterpart, and I'm not sure it's worth blocking this based on the cli.

Comment (munkyshi) [reply pyproject.toml]: Ah, I see: https://docs.kolena.com/workflow/advanced-usage/packaging-for-automated-evaluation/ -- do we need to preserve this capability?

Comment (munkyshi) [reply pyproject.toml]: Actually, this seems to work out of the box with uv!
```
❯ uv run kolena --help
Ignoring existing lockfile due to change in supported environments
Usage: kolena [OPTIONS] COMMAND [ARGS]...

Options:
  -h, --help  Show this message and exit.

Commands:
  evaluator
  repository
```

Comment (yifanwu-kolena) [reply pyproject.toml]: this is cool, but how can uv know the cli entrypoint? 🤔 

based on [the doc](https://github.com/astral-sh/uv/blob/039b68367e6481690ca001c44e61500bd1e42570/docs/concepts/projects.md?plain=1#L210-L255), we can define `[project.scripts]`

Comment (munkyshi) [reply pyproject.toml]: Hmm, I wonder if there was something leftover from my local state with `[tool.poetry.scripts]`

Nice catch, re-added as `[project.scripts]`

Comment (yifanwu-kolena) [reply pyproject.toml]: maybe there are some heuristic magic behind the scene, but i believe explicit is better than implicit

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -38,25 +38,24 @@ jobs:
...
       - name: Fetch tags to enable autoversioning
         run: git fetch --prune --unshallow --tags
 
-      - name: Update package version to PEP 440-compliant production release tag
-        run: poetry version $(git describe --tags --abbrev=0)
-
-      - name: Install Poetry dependencies
-        run: poetry install
+      - name: Install dependencies
+        run: uv pip install --all-extras -r pyproject.toml
```
Comment (yifanwu-kolena) [inline .github/workflows/docs.yml:50]: does this handle the dynamic release tag in uv? my concern is that we might publish the version as `0.999.0`

File: docs/automations/set-up-natural-language-search.md [hunk 3]
```diff
@@ -70,16 +70,11 @@ export KOLENA_TOKEN="********"
...
-    ```shell
-    poetry source add --priority=supplemental kolena-embeddings "https://gateway.kolena.cloud/repositories"
-    ```
-
-    Then run the following command, making sure to replace <KOLENA_TOKEN> with the token retrieved from the developer page:
+    Run the following command, making sure to replace <KOLENA_TOKEN> with the token retrieved from the developer page:
     ```shell
-    poetry config http-basic.kolena-embeddings <KOLENA_TOKEN> ""
+    uv add --extra-index-url="https://<KOLENA_TOKEN>@gateway.kolena.cloud/repositories" kolena-embeddings
     ```
```
Comment (yifanwu-kolena) [inline docs/automations/set-up-natural-language-search.md:78]: just want to verify that we tested this manually already

**Review by yifanwu-kolena — APPROVED** [review]
🚢

File: docs/automations/set-up-natural-language-search.md (hunk 3 above)
Comment (munkyshi) [reply docs/automations/set-up-natural-language-search.md:78]: Yep, I ran this manually and was able to install `kolena-embeddings`

**Pull request merged**